### PR TITLE
[SYCL] Reverse logic of how we add SPIR-V extensions

### PR DESCRIPTION
**Pull request by MrSidims**
New extensions no longer passed to the translator by default, so
we can wait until they are supported by devices' SPIR-V consumers.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: clang/test/Driver/sycl-offload.c [hunk 1]
```diff
@@ -666,10 +666,10 @@
...
...-TOOLS-CPU: llvm-spirv{{.*}} "-o" "[[OUTPUT3:.+\.spv]]" "-spirv-max-version=1.3" "-spirv-debug-info-version=legacy" "-spirv-allow-extra-diexpressions" "-spirv-allow-unknown-intrinsics=llvm.genx." "-spirv-ext=+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max,+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls,+SPV_INTEL_subgroups,+SPV_INTEL_media_block_io,+SPV_INTEL_device_side_avc_motion_estimation,+SPV_INTEL_fpga_loop_controls,+SPV_INTEL_fpga_memory_attributes,+SPV_INTEL_fpga_memory_accesses,+SPV_INTEL_unstructured_loop_controls,+SPV_INTEL_fpga_reg,+SPV_INTEL_blocking_pipes,+SPV_INTEL_function_pointers,+SPV_INTEL_kernel_attributes,+SPV_INTEL_io_pipes,+SPV_INTEL_inline_assembly,+SPV_INTEL_arbitrary_precision_integers,+SPV_INTEL_optimization_hints,+SPV_INTEL_float_controls2,+SPV_INTEL_vector_compute,+SPV_INTEL_fast_composite,+SPV_INTEL_fpga_buffer_location,+SPV_INTEL_arbitrary_precision_fixed_point,+SPV_INTEL_arbitrary_precision_floating_point,+SPV_INTEL_arbitrary_precision_floating_point,+SPV_INTEL_variable_length_array,+SPV_INTEL_fp_fast_math_mode,+SPV_INTEL_fpga_cluster_attributes,+SPV_INTEL_loop_fuse,+SPV_INTEL_long_constant_composite" "[[OUTPUT2_1]]"
```
Comment (AlexeySachkov) [inline clang/test/Driver/sycl-offload.c:669]: I think that we should probably have a separate single test for extensions string passed to llvm-spirv and not overcomplicate existing tests

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -8446,28 +8446,46 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
...
     TranslatorArgs.push_back("-spirv-allow-unknown-intrinsics=llvm.genx.");
 
-    // Disable SPV_INTEL_usm_storage_classes by default since it adds new
-    // storage classes that represent global_device and global_host address
-    // spaces, which are not supported for all targets. With the extension
-    // disable the storage classes will be lowered to CrossWorkgroup storage
-    // class that is mapped to just global address space. The extension is
-    // supposed to be enabled only for FPGA hardware.
-    std::string ExtArg("-spirv-ext=+all,-SPV_INTEL_usm_storage_classes");
+    std::string ExtArg("-spirv-ext=");
```
Comment (AlexeySachkov) [inline clang/lib/Driver/ToolChains/Clang.cpp:8449]: To be even more sure, I suggest to add `-all` as the first value here

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 3]
```diff
@@ -8446,28 +8446,46 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
...
+    std::string DefaultExtArg =
+        "+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max,"
+        "+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls";
```
Comment (AlexeySachkov) [inline clang/lib/Driver/ToolChains/Clang.cpp:8452]:
```suggestion
        "+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max"
        ",+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls";
```

To be consistent with the lines below

File: clang/test/Driver/sycl-offload.c (hunk 1 above)
Comment (MrSidims) [reply clang/test/Driver/sycl-offload.c]: Applied the suggestions, thanks!

**Review by AlexeySachkov — DISMISSED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Review by AGindinson — APPROVED** [review]
LGTM! The single, separate LIT file for full SPIR-V extension testing is a really nice find.

**Pull request merged**